rmw_microxrcedds_c/src/rmw_publisher.c

```diff
@@ -235,10 +241,40 @@ rmw_publisher_count_matched_subscriptions(
   const rmw_publisher_t * publisher,
   size_t * subscription_count)
 {
+#ifdef RMW_UXRCE_GRAPH
+  rmw_ret_t ret = RMW_RET_OK;
+  const char * topic_name = publisher->topic_name;
+  rcutils_allocator_t allocator = rcutils_get_default_allocator();
+
+  rmw_uxrce_publisher_t * custom_publisher = (rmw_uxrce_publisher_t *)publisher->data;
+  const rmw_node_t * node = custom_publisher->owner_node->rmw_handle;
+
+  rmw_topic_endpoint_info_array_t subscriptions_info =
+    rmw_get_zero_initialized_topic_endpoint_info_array();
+
+  if (RMW_RET_OK != rmw_get_subscriptions_info_by_topic(
+      node,
+      &allocator,
+      topic_name,
+      false,
+      &subscriptions_info)) {
+    ret = RMW_RET_ERROR;
+    goto pub_count_sub_fail;
+  } else {
+    *subscription_count = subscriptions_info.size;
+  }
+
+pub_count_sub_fail:
+  if (RMW_RET_OK != rmw_topic_endpoint_info_array_fini(&subscriptions_info, &allocator)) {
+    ret = RMW_RET_ERROR;
+  }
```

Comment:
If this fails we return OK?

```suggestion
    rmw_topic_endpoint_info_array_fini(&subscriptions_info, &allocator);
    ret = RMW_RET_ERROR;
```

Reply:
notice that when we jump here, we are setting previously the `ret` variable to `RMW_RET_ERROR`. We are returning OK just in the normal behaviour case (that is, the jump coded in 262 never happening and also the resources freed with `rmw_topic_endpoint_info_array_fini` returning RMW_RET_OK).

Reply:
Cool, closing this